Softeq.XToolkit.PushNotifications.iOS/Services/IosPushNotificationsConsumer.cs

```diff
@@ -0,0 +1,248 @@
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public bool TryHandleNotificationResponse(
+            UNUserNotificationCenter center,
+            UNNotificationResponse response,
+            Action completionHandler)
+        {
+            var actionIdentifier = response.ActionIdentifier;
+            var userInfo = response.Notification.Request.Content.UserInfo;
+
+            if (!_pushNotificationsParser.TryParse(userInfo, out var parsedNotification))
+            {
+                return false;
+            }
+
+            if (response.IsCustomAction)
+            {
+                var textInput = (response as UNTextInputNotificationResponse)?.UserText ?? string.Empty;
+                _notificationCategoriesProvider.HandlePushNotificationCustomAction(parsedNotification, actionIdentifier, textInput);
+            }
+            else if (response.IsDefaultAction)
+            {
+                if (parsedNotification.IsSilent)
+                {
+                    _pushNotificationsHandler.HandleSilentPushNotification(parsedNotification);
+                }
+                else
+                {
+                    _pushNotificationsHandler.HandlePushNotificationTapped(parsedNotification);
+                }
+            }
+            else if (response.IsDismissAction)
+            {
+                _notificationCategoriesProvider.HandlePushNotificationCustomAction(parsedNotification, actionIdentifier, string.Empty);
+            }
+
+            completionHandler.Invoke();
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public bool TryHandleRemoteNotification(
+            UIApplication application,
+            NSDictionary userInfo,
+            Action<UIBackgroundFetchResult> completionHandler)
+        {
+            // do nothing?
+            return false;
+        }
+
+        /// <inheritdoc />
+        public void OnPushNotificationAuthorizationResult(bool isGranted)
+        {
+            _pushNotificationsHandler.OnPushPermissionsRequestCompleted(isGranted);
+        }
+
+        /// <inheritdoc />
+        public void OnRegisteredForRemoteNotifications(UIApplication application, NSData deviceToken)
+        {
+            var token = ParseDeviceToken(deviceToken);
+
+            OnRegisteredForPushNotifications(token);
+        }
+
+        /// <inheritdoc />
+        public void OnFailedToRegisterForRemoteNotifications(UIApplication application, NSError error)
+        {
+            _logger.Warn($"Push Notifications failed to register: {error.Description}");
+            OnRegisterFailedInternal().FireAndForget(_logger);
+        }
+
+        /// <inheritdoc />
+        public Task OnUnregisterFromPushNotifications()
+        {
+            return UnregisterFromRemotePushNotifications();
+        }
+
+        private static string ParseDeviceToken(NSData deviceToken)
+        {
+            return deviceToken.ToString().Replace("-", string.Empty);
```

Comment:
Broken parsing!
Please use `.AsString()` instead of `toString()`

Reply:
Fixed